routing/endpointregistry.go

```diff
@@ -86,14 +86,21 @@ func (r *EndpointRegistry) Do(routes []*Route) []*Route {
 
 	for _, route := range routes {
 		if route.BackendType == eskip.LBBackend {
-			for _, epi := range route.LBEndpoints {
-				metrics := r.GetMetrics(epi.Host)
-				if metrics.DetectedTime().IsZero() {
-					metrics.SetDetected(now)
+			for i := range route.LBEndpoints {
+				epi := &route.LBEndpoints[i]
+				epi.Metrics = r.GetMetrics(epi.Host)
+				if epi.Metrics.DetectedTime().IsZero() {
+					epi.Metrics.SetDetected(now)
 				}
 
-				metrics.SetLastSeen(now)
+				epi.Metrics.SetLastSeen(now)
 			}
+		} else {
```

Comment:
@RomanZavodskikh Here it should be `else if route.BackendType == eskip.NetworkBackend` because there are other types for which `route.Host` will be empty and `r.GetMetrics(route.Host)` will create an entry for empty key.

Please add this condition and a test for `func (r *EndpointRegistry) Do(routes []*Route) []*Route`

Reply:
Very nice catch. :-)
Addressed it here: https://github.com/zalando/skipper/pull/2869